### PR TITLE
[oneDPL][examples] + fix a error in histogram example

### PR DESCRIPTION
**Pull request by MikeDvorskiy**
[oneDPL][examples] + fix a error in histogram example. A wrong size of passed sequence.
"The sum of accessRange and accessOffset should not exceed the range of buffer."

### CODE REVIEW

File: include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h [hunk 1]
```diff
@@ -79,6 +79,18 @@ class all_view
         cgh.require(m_acc);
     }
 
+  private:
+    accessor_t create_accessor(sycl::buffer<_T, 1> __buf, diff_type __offset, diff_type __n)
```
Comment (rarutyun) [inline include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h:83]: it is dangerous to call non-static function in initialization. As I see, you don't need `this`, so please make it static.

Comment (rarutyun) [inline include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h:83]: Uglification: `__create_accessor`. Actually this comment is applicable to many things in `all_view`: `__accessor_t`, `m_acc`, `return_t`, etc.

Comment (MikeDvorskiy) [reply include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h:83]: Actually, I tried wo be consistent with whole `class all_view`.

File: examples/histogram/src/main.cpp [hunk 2]
```diff
@@ -90,7 +90,7 @@ void sparse_histogram(std::vector<uint64_t> &input) {
 
   std::cout << "success for Sparse Histogram:\n";
   std::cout << "[";
-  for (int i = 0; i < num_bins - 1; i++) {
+  for (int i = 0; i < num_bins; i++) {
```
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply examples/histogram/src/main.cpp:93]: discussed ofline

File: examples/histogram/src/main.cpp [hunk 3]
```diff
@@ -70,7 +70,7 @@ void sparse_histogram(std::vector<uint64_t> &input) {
 
   auto num_bins = std::transform_reduce(
       oneapi::dpl::execution::dpcpp_default, oneapi::dpl::begin(histogram_buf),
-      oneapi::dpl::end(histogram_buf), oneapi::dpl::begin(histogram_buf) + 1, 1,
+      oneapi::dpl::end(histogram_buf) - 1, oneapi::dpl::begin(histogram_buf) + 1, 1,
```
[the remark this answers is not in the document]
Comment (MikeDvorskiy) [reply examples/histogram/src/main.cpp:73]: discussed offline

File: include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h [hunk 4]
```diff
@@ -32,40 +32,40 @@ namespace __ranges
 {
 
 //A SYCL range over SYCL buffer
-template <typename _T, sycl::access::mode AccMode = sycl::access::mode::read,
+template <typename _T, sycl::access::mode __AccMode = sycl::access::mode::read,
```
Comment (rarutyun) [inline include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h:35]:
```suggestion
template <typename _T, sycl::access::mode _AccMode = sycl::access::mode::read,
```

**Review by rarutyun — APPROVED** [review]

**Pull request merged**